### PR TITLE
[FLINK-18951][python][docs] Add documentation for Configurations in Python DataStream API.

### PR DESCRIPTION
**Pull request by shuiqiangchen**
## What is the purpose of the change

Add documentation for Configurations in Python DataStream API.

## Brief change log

- Added a new page named dependency_management.md under dev/python/datastream

## Verifying this change

This pull request added documentations for existing Python DataStream API, which has no need for test case coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? ( not documented)

### CODE REVIEW

File: docs/dev/python/user-guide/datastream/dependency_management.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        <p>Specifies a requirements.txt file which defines the third-party dependencies. These dependencies will be installed to a temporary directory and added to the PYTHONPATH of the python UDF worker. For the dependencies which could not be accessed in the cluster, a directory which contains the installation packages of these dependencies could be specified using the parameter "requirements_cached_dir". It will be uploaded to the cluster to support offline installation.</p>
+{% highlight python %}
+# commands executed in shell
+echo numpy==1.16.5 > requirements.txt
+pip download -d cached_dir -r requirements.txt --no-binary :all:
+
+# python code
+stream_execution_environment.set_python_requirements("requirements.txt", "cached_dir")
+{% endhighlight %}
+        <p>Please make sure the installation packages matches the platform of the cluster and the python version used. These packages will be installed using pip, so also make sure the version of Pip (version >= 7.1.0) and the version of SetupTools (version >= 37.0.0).</p>
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/datastream/dependency_management.md:60]: SetupTools -> `Setuptools`

File: docs/dev/python/user-guide/table/dependency_management.zh.md [hunk 2]
```diff
@@ -61,49 +61,49 @@ table_env.add_python_file(file_path)
...
+# shell 命令
 echo numpy==1.16.5 > requirements.txt
 pip download -d cached_dir -r requirements.txt --no-binary :all:
 
-# python code
+# Python 代码
 table_env.set_python_requirements("requirements.txt", "cached_dir")
 {% endhighlight %}
-        <p>Please make sure the installation packages matches the platform of the cluster and the python version used. These packages will be installed using pip, so also make sure the version of Pip (version >= 7.1.0) and the version of SetupTools (version >= 37.0.0).</p>
+        <p>请确保这些依赖安装包和集群运行环境以及 Python 版本相匹配。此外，这些依赖将通过 Pip 安装， 请确保 Pip的版本（version >= 7.1.0） 和 Setup Tools 的版本（version >= 37.0.0）和依赖安装包兼容。</p>
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/table/dependency_management.zh.md:73]: SetupTools -> Setuptools

File: docs/dev/python/user-guide/datastream/dependency_management.zh.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        <p>配置一个 requirements.txt 文件用于指定 Python 第三方依赖，这些依赖会被安装到一个临时目录并添加到 Python Worker 的 PYTHONPATH 中。对于在集群中无法访问的外部依赖，用户可以通过 "requirements_cached_dir" 参数指定一个包含这些依赖安装包的目录，这个目录文件会被上传到集群并实现离线安装。</p>
+{% highlight python %}
+# shell 命令
+echo numpy==1.16.5 > requirements.txt
+pip download -d cached_dir -r requirements.txt --no-binary :all:
+
+# Python 代码
+stream_execution_environment.set_python_requirements("requirements.txt", "cached_dir")
+{% endhighlight %}
+        <p>请确保这些依赖安装包和集群运行环境以及 Python 版本相匹配。此外，这些依赖将通过 Pip 安装， 请确保 Pip的版本（version >= 7.1.0） 和 Setup Tools 的版本（version >= 37.0.0）和依赖安装包兼容。</p>
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/datastream/dependency_management.zh.md:58]: SetupTools -> `Setuptools`

File: docs/dev/python/user-guide/table/dependency_management.zh.md [hunk 4]
```diff
@@ -61,49 +61,49 @@ table_env.add_python_file(file_path)
...
-        <p>Specifies a requirements.txt file which defines the third-party dependencies. These dependencies will be installed to a temporary directory and added to the PYTHONPATH of the python UDF worker. For the dependencies which could not be accessed in the cluster, a directory which contains the installation packages of these dependencies could be specified using the parameter "requirements_cached_dir". It will be uploaded to the cluster to support offline installation.</p>
+        <p>配置一个 requirements.txt 文件用于指定 Python 第三方依赖，这些依赖会被安装到一个临时目录并添加到 Python Worker 的 PYTHONPATH 中。对于在集群中无法访问的外部依赖，用户可以通过 "requirements_cached_dir" 参数指定一个包含这些依赖安装包的目录，这个目录文件会被上传到集群并实现离线安装。</p>
 {% highlight python %}
-# commands executed in shell
+# shell 命令
 echo numpy==1.16.5 > requirements.txt
 pip download -d cached_dir -r requirements.txt --no-binary :all:
 
-# python code
+# Python 代码
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/table/dependency_management.zh.md:70]: Python 代码 -> python 代码

File: docs/dev/python/user-guide/table/dependency_management.zh.md [hunk 5]
```diff
@@ -61,49 +61,49 @@ table_env.add_python_file(file_path)
     <tr>
       <td><strong>set_python_requirements(requirements_file_path, requirements_cache_dir=None)</strong></td>
       <td>
-        <p>Specifies a requirements.txt file which defines the third-party dependencies. These dependencies will be installed to a temporary directory and added to the PYTHONPATH of the python UDF worker. For the dependencies which could not be accessed in the cluster, a directory which contains the installation packages of these dependencies could be specified using the parameter "requirements_cached_dir". It will be uploaded to the cluster to support offline installation.</p>
+        <p>配置一个 requirements.txt 文件用于指定 Python 第三方依赖，这些依赖会被安装到一个临时目录并添加到 Python Worker 的 PYTHONPATH 中。对于在集群中无法访问的外部依赖，用户可以通过 "requirements_cached_dir" 参数指定一个包含这些依赖安装包的目录，这个目录文件会被上传到集群并实现离线安装。</p>
 {% highlight python %}
-# commands executed in shell
+# shell 命令
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/table/dependency_management.zh.md:66]: 执行下面的shell命令

**Review by HuangXingBo — COMMENTED** [review]
@shuiqiangchen Thanks a lot for adding the doc. Because most of the content in the document is the same as the dependency management document of the Python Table API, I only left some comments.

**Comment by shuiqiangchen** [comment]
Hi @HuangXingBo , highly appreciate for your suggestions, I have updated the PR, please have a look at it.

File: docs/dev/python/user-guide/table/dependency_management.zh.md [hunk 6]
```diff
@@ -61,49 +61,49 @@ table_env.add_python_file(file_path)
...
+# 执行下面的 shell 命令
 echo numpy==1.16.5 > requirements.txt
 pip download -d cached_dir -r requirements.txt --no-binary :all:
 
-# python code
+# python 代码
 table_env.set_python_requirements("requirements.txt", "cached_dir")
 {% endhighlight %}
-        <p>Please make sure the installation packages matches the platform of the cluster and the python version used. These packages will be installed using pip, so also make sure the version of Pip (version >= 7.1.0) and the version of SetupTools (version >= 37.0.0).</p>
+        <p>请确保这些依赖安装包和集群运行环境以及 Python 版本相匹配。此外，这些依赖将通过 Pip 安装， 请确保 Pip的版本（version >= 7.1.0） 和 Setuptools 的版本（version >= 37.0.0）和依赖安装包兼容。</p>
```
Comment (HuangXingBo) [inline docs/dev/python/user-guide/table/dependency_management.zh.md:73]:
```suggestion
        <p>请确保这些依赖安装包和集群运行环境所使用的的 Python 版本相匹配。此外，这些依赖将通过 Pip 安装， 请确保 Pip的版本（version >= 7.1.0） 和 Setuptools 的版本（version >= 37.0.0）符合要求。</p>
```

**Review by HuangXingBo — COMMENTED** [review]
@shuiqiangchen Thanks a lot for the update. LGTM. I only left a minor comment

**Comment by shuiqiangchen** [comment]
@HuangXingBo Thank you for your comments, I have updated the PR according to your suggestions, please have a look.

**Comment by HuangXingBo** [comment]
@shuiqiangchen Thanks a lot for the update. LGTM

**Comment by dianfu** [comment]
@shuiqiangchen As StreamExecutionEnvironment#add_jars has been merged. Could you rebase the PR and add documentation on how to specify Java dependencies in Python DataStream jobs?

**Comment by shuiqiangchen** [comment]
@dianfu Thank you for your reminding, I have added the java dependency management part for Python DataStream API docs.

File: docs/dev/python/datastream-api-users-guide/dependency_management.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+# Use the add_classpaths() to add the dependent jars URL into the classpath. And the URL will also be added to the classpath of the cluster.
+# NOTE: The Paths must specify a protocol (e.g. file://) and users should ensure that the URLs are accessible on both the client and the cluster.
+stream_execution_environment.add_classpaths("file:///my/jar/path/connector.jar", ...)
+{% endhighlight %}
+**Note** These APIs can be called multiple times.
+
+# Python Dependency
+
+If third-party Python dependencies are used, Users can specify the dependencies with the following Python DataStream
```
Comment (dianfu) [inline docs/dev/python/datastream-api-users-guide/dependency_management.md:43]:
```suggestion
If third-party Python dependencies are used, you can specify the dependencies with the following Python DataStream 
```

File: docs/dev/python/datastream-api-users-guide/dependency_management.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+{% highlight python %}
+# Use the add_jars() to add local jars and the jars will be uploaded to the cluster.
+# NOTE: Only local file URLs (start with "file://") are supported.
+stream_execution_environment.add_jars("file:///my/jar/path/connector.jar", ...)
+
+# Use the add_classpaths() to add the dependent jars URL into the classpath. And the URL will also be added to the classpath of the cluster.
+# NOTE: The Paths must specify a protocol (e.g. file://) and users should ensure that the URLs are accessible on both the client and the cluster.
+stream_execution_environment.add_classpaths("file:///my/jar/path/connector.jar", ...)
+{% endhighlight %}
+**Note** These APIs can be called multiple times.
```
Comment (dianfu) [inline docs/dev/python/datastream-api-users-guide/dependency_management.md:39]:
```suggestion
**Note:** These APIs can be called multiple times.
```

**Review by dianfu — APPROVED** [review]
LGTM. I will integrate the following comments during merging.

**Pull request closed without merging**